core/array/shared/slice.rb

```diff
@@ -533,6 +533,25 @@ def to.to_int() -2 end
     a.send(@method, eval("(-9...)")).should == nil
   end
 
+  ruby_version_is "3.0" do
+    it "can be sliced with Enumerator::ArithmeticSequence" do
+      a = [0, 1, 2, 3, 4, 5]
+      a.send(@method, eval("(0..).step(2)")).should == [0, 2, 4]
+      a.send(@method, eval("(0..).step(3)")).should == [0, 3]
+      a.send(@method, eval("(0..).step(10)")).should == [0]
+      a.send(@method, eval("(0..1).step(2)")).should == [0]
+      a.send(@method, eval("(..0).step(1)")).should == [0]
+      a.send(@method, eval("(...0).step(1)")).should == []
+      a.send(@method, eval("(2..).step(2)")).should == [2, 4]
+      a.send(@method, eval("(2..-2).step(2)")).should == [2, 4]
+      a.send(@method, eval("(2...-2).step(2)")).should == [2]
+      a.send(@method, eval("(2..).step(-2)")).should == [2, 0]
+      a.send(@method, eval("(2..).step(-3)")).should == [2]
```

Comment:
It would be useful to separate these examples into a separate `it` and use the description to describe what it does.
Could you do that?

Reply:
@eregon Sure, will do!